src/util/system.cpp

```diff
@@ -372,6 +373,86 @@ bool ArgsManager::IsArgSet(const std::string& strArg) const
     return !GetSetting(strArg).isNull();
 }
 
+static bool GetSettingsPath(fs::path* filepath = nullptr, bool temp = false)
+{
+    if (gArgs.IsArgNegated("-settings")) return false;
+    if (filepath) {
+        std::string settings = gArgs.GetArg("-settings", BITCOIN_SETTINGS_FILENAME);
+        *filepath = fs::absolute(temp ? settings + ".tmp" : settings, GetDataDir(/* net_specific= */ true));
+    }
+    return true;
+}
+
+bool ArgsManager::EnableSettingsFile()
+{
+    return GetSettingsPath();
+}
+
+bool ArgsManager::ReadSettingsFile()
+{
+    fsbridge::ifstream file;
+    fs::path filepath;
+    if (!GetSettingsPath(&filepath, /* temp= */ false)) return true;
+    file.open(filepath);
+    if (!file.is_open()) return true; // Ok for file not to exist.
+
+    util::SettingsValue in;
+    if (!in.read(std::string{std::istreambuf_iterator<char>(file), std::istreambuf_iterator<char>()})) {
+        LogPrintf("Error: Unable to parse settings file %s\n", filepath.string());
```

Comment:
These logprints are currently going nowhere because `InitLogging()` is called after `ReadSettingsFile()`. Not saying it is a problem.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15935#discussion_r435743918

> These logprints are currently going nowhere because `InitLogging()` is called after `ReadSettingsFile()`. Not saying it is a problem.

Good catch, errors are returned now